searchcore/src/vespa/searchcore/proton/matching/match_master.cpp

```diff
@@ -75,7 +80,8 @@ MatchMaster::match(const MatchParams &params,
                 ? static_cast<IMatchLoopCommunicator&>(timedCommunicator)
                 : static_cast<IMatchLoopCommunicator&>(communicator);
         threadState.emplace_back(std::make_unique<MatchThread>(i, threadBundle.size(), params, mtf, com, *scheduler,
-                                                               resultProcessor, mergeDirector, distributionKey));
+                                                               resultProcessor, mergeDirector, distributionKey,
+                                                               request.getRelativeTime(), request.trace().getLevel()));
```

Comment:
it looks like thread-specific traces will inherit the global start time/time concept(?) this would skew the internal relative thread-specific timestamps and also yield a duration that is too high.

Reply:
The intention was that all timestamps should be relative to the start of the query. However adding duration to the thread traces made that a bad idea.
I think it is better to follow the pattern used for tracing in the qrs that each timestamp is relative to start of the request. All federations also relate to the global start time when adding their timestamps.
So I think it is better to drop the notion of duration from the per thread traces.